File: JavaScript/150-Evaluate Reverse Polish Notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack = [];
    let operators = new Set(["+","-","*","/"]);
    
    for (let token of tokens){
        if (!operators.has(token)){
            stack.push(parseInt(token));
        }else{
            let num2 = stack.pop();
            let num1 = stack.pop();
            if (token === "+"){
                stack.push(num1+num2);
            }else if (token === "-"){
                stack.push(num1-num2);
            }else if (token === "*"){
                stack.push(num1*num2);
            }else{
                stack.push(parseInt(num1/num2));
            }
        }
    }
    return stack[0];
};

File: C++/150-Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;
        unordered_set<string> s;
        for (char c:"+-*/") s.insert(string(1, c));
        
        for (string t:tokens) {
            if (s.find(t) != s.end()) {
                int t2 = stk.top(); stk.pop();
                int t1 = stk.top(); stk.pop();
                if (t == "+") {
                    stk.push(t1+t2);
                } else if (t == "-") {
                    stk.push(t1-t2);
                } else if (t == "*") {
                    stk.push(t1*t2);
                } else {
                    stk.push(t1/t2);
                }
            } else {
                stk.push(stoi(t));
            }
        }
        return int(stk.top());
    }
};